### DISCUSSION TITLE
Memory Usage Extremely Heavy when processing large EPG

### DISCUSSION THREAD

**Discussion by jamesarbrown** [euzu/tuliprox#73] [Ideas]
m3ufilter happily works on <100MB of memory most of the time.

But as it processes the JSON EPG, I have had to allocate 2GB, which to me is too much

Is there anyway to part process, maybe stream incoming EPG to file etc as I think its only mirroring it from source?

**Comment by euzu** [comment]
There was a feature to implement epg shift user based. I have processed the epg with a sax parser. 
I could try to avoid loading the epg into memory and process it with the sax parser. 

I have created a future request [issue-86](https://github.com/euzu/m3u-filter/issues/86)

**Comment by euzu** [comment]
This should.be resolved with the latest release.
@jamesarbrown can you confirm?

**Comment by euzu** [comment]
closed due inactivity
